### DISCUSSION TITLE
Is there any possibility in using Pascal Case for all column names?

### DISCUSSION THREAD

**Discussion by ervansanjaya** [spatie/laravel-permission#2313] [Q&A] [answered]
Hi everyone, I was seeking for the answer on the documentation but seems got myself lost.

I am building a fresh app which requires table and column name to be PascalCased. In that sense I need to modify the default table name like:

Roles:
**Columns:** Id|Name|GuardName|CreatedAt|CreatedBy|UpdatedAt|UpdatedBy|

Permissions:

ModelHasPermissions:
RoleHasPermissions:
ModelHasRoles:

I found out how to change the primary key, the reference key RoleId. But can I also upgrade the default snake_cased name as guard_name to be GuardName from the configuration?

Thank you

**Comment by erikn69** [comment, marked as answer]
You can't change `name`, `guard_name` at the moment, and I don't think it will change in the near future

**Reply by ervansanjaya** [reply]
Hi @erikn69 thanks for the reply.
Perharps I can do a 'hacky' way to achieve this? Like overriding through the model?
Or maybe there's another way I can do with Pascal Cased columns?

**Reply by erikn69** [reply]
> Perharps I can do a 'hacky' way to achieve this? Like overriding through the model?

You could override, but, I don't know if it will be possible to have all the functionalities that way, no one has ever required something like this

**Reply by ervansanjaya** [reply]
Alright then @erikn69 I prefer not to force myself to overwrite this. Thanks for your response!

**Reply by erikn69** [reply]
Mark this as answered if you dont have more questions
